src/lib/swapper/swappers/ThorchainSwapper/utils/getUniqueAddressSubstring.test.ts

```diff
@@ -0,0 +1,40 @@
+import { describe, expect, it } from 'vitest'
+
+import { getUniqueAddressSubstring } from './getUniqueAddressSubstring'
+
+const FINAL_ASSET_ADDRESS = '0x8a65ac0E23F31979db06Ec62Af62b132a6dF4741'
+
+const COLLIDING_FINAL_ASSET_ADDRESS = '0x8a65ac0E23F31979db06Ec62Af62b132i6dF4741'
+
+const BIGGER_COLLIDING_FINAL_ASSET_ADDRESS = '0x8a65ac0E23F31979db06Ec62Af62b432a6dF4741'
+const BIGGER_COLLIDING_FINAL_ASSETID = `eip155:1/erc20:${BIGGER_COLLIDING_FINAL_ASSET_ADDRESS}`
+
+const SHORTENED_FINAL_ASSET_ADDRESS = 'a6df4741'
+const FINAL_ASSETID = `eip155:1/erc20:${FINAL_ASSET_ADDRESS}`
+const COLLIDING_FINAL_ASSETID = `eip155:1/erc20:${COLLIDING_FINAL_ASSET_ADDRESS}`
```

Comment:
This isn't a valid `AssetId` since `COLLIDING_FINAL_ASSET_ADDRESS` is checksummed - see refactored tests and util in https://gist.github.com/gomesalexandre/ad4450bec75c4c7d907956ee976067aa